treeherder/model/models.py

```diff
@@ -199,42 +199,59 @@ def __str__(self):
     @classmethod
     def search(cls, search_term):
         max_size = 50
+
         # 90 days ago
         time_limit = datetime.datetime.now() - datetime.timedelta(days=90)
+
         # Wrap search term so it is used as a phrase in the full-text search.
-        search_term_fulltext = '"%s"' % search_term.replace("\"", "")
+        search_term_fulltext = '"%s"' % search_term.replace("\"", "").replace("-", "")
+
         # Substitute escape and wildcard characters, so the search term is used
         # literally in the LIKE statement.
-        search_term_like = search_term.replace('=', '==').replace(
-            '%', '=%').replace('_', '=_')
-        recent = cls.objects.raw(
-            '''
+        search_term_like = search_term.replace('=', '==').replace('%', '=%').replace('_', '=_')
+
+        recent_qs = cls.objects.raw(
+            """
             SELECT id, summary, crash_signature, keywords, os, resolution, status,
-            MATCH (`summary`) AGAINST (%s IN BOOLEAN MODE) AS relevance
-            FROM bugscache
-            WHERE 1
-              AND resolution = ''
-              AND `summary` LIKE CONCAT ('%%%%', %s, '%%%%') ESCAPE '='
-              AND modified >= %s
-            ORDER BY relevance DESC
-            LIMIT 0,%s
-            ''', [search_term_fulltext, search_term_like, time_limit,
-                  max_size])
-
-        all_others = cls.objects.raw(
-            '''
+             MATCH (`summary`) AGAINST (%s IN BOOLEAN MODE) AS relevance
+              FROM bugscache
+             WHERE 1
+               AND resolution = ''
+               AND `summary` LIKE CONCAT ('%%%%', %s, '%%%%') ESCAPE '='
+               AND modified >= %s
+          ORDER BY relevance DESC
+             LIMIT 0,%s
+            """,
+            [search_term_fulltext, search_term_like, time_limit, max_size],
+        )
+
+        try:
+            open_recent = [model_to_dict(item, exclude=["modified"]) for item in recent_qs]
+        except Exception as e:
```

Comment:
So the only downside to this is that if there is a temporary infra issue (eg `OperationalError`), the empty results will then be cached in `get_error_summary()`. As such I think even if we don't parse the message/error code for a specific error, we should still catch on `ProgrammingError` instead of `Exception`.

Reply:
Good point :)